src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/index.tsx

```diff
@@ -60,17 +70,23 @@ type Props = {
 } & RouteComponentProps<{orgId: string; projectId: string; ruleId: string}, {}>;
 
 type State = {
-  rule: IssueAlertRule | null;
+  rule: UnsavedIssueAlertRule | IssueAlertRule;
   loading: boolean;
   error: null | {
     [key: string]: string[];
   };
   environments: Environment[];
 };
 
+function isSavedAlertRule(
+  rule: UnsavedIssueAlertRule | IssueAlertRule
+): rule is IssueAlertRule {
```

Comment:
Cool syntax 🧐Learned something new today

Reply:
This is called a typeguard: https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types